SetupDataPkg/ConfProfileMgrDxe/ConfProfileMgrDxe.c

```diff
@@ -0,0 +1,248 @@
+  EFI_GUID    CachedProfile;
+  BOOLEAN     NeedToFreeMem = TRUE;
+
+  // RetrieveActiveProfileGuid will validate the profile guid is valid before returning
+  Status = RetrieveActiveProfileGuid (&ActiveProfileGuid);
+
+  // Check if the status failed, but don't error out, try to use cached profile
+  if (EFI_ERROR (Status) || (ActiveProfileGuid == NULL)) {
+    NeedToFreeMem = FALSE;
+    DEBUG ((DEBUG_ERROR, "%a RetrieveActiveProfileGuid failed (%r)! Attempting to set cached profile\n", __FUNCTION__, Status));
+
+    Status = gRT->GetVariable (
+                    CACHED_CONF_PROFILE_VARIABLE_NAME,
+                    &gConfProfileMgrVariableGuid,
+                    &Attributes,
+                    &Size,
+                    &CachedProfile
+                    );
+
+    if (EFI_ERROR (Status) || (Size != sizeof (EFI_GUID))) {
+      DEBUG ((DEBUG_ERROR, "%a failed to read cached profile, defaulting to default profile (%r)!\n", __FUNCTION__, Status));
+      ActiveProfileGuid = (EFI_GUID *)&gSetupDataPkgGenericProfileGuid;
+    } else {
+      ActiveProfileGuid = &CachedProfile;
+    }
+  }
+
+  // Set the PCD for the profile parser lib to read during profile validation and for the ConfApp
+  Status = PcdSetPtrS (PcdSetupConfigActiveProfileFile, &Size, ActiveProfileGuid);
+
+  // If we fail to set the PCD, the system will still be in the correct profile, but the ConfApp will report that
+  // the generic profile is active, so we still want to validate the active profile
+  if (EFI_ERROR (Status) || (Size != sizeof (EFI_GUID))) {
+    DEBUG ((DEBUG_ERROR, "%a failed to set ActiveProfile PCD (%r)!\n", __FUNCTION__, Status));
+    ASSERT (FALSE);
+  }
+
+  // We have the active profile guid, we need to load it and validate the contents against flash.
+  // ValidateProfile does not return a status, in case of failure, it writes the chosen profile to flash
+  // and resets the system. Only validate the profile if we are in CUSTOMER_MODE (not manufacturing mode),
+  // as in debug and bringup scenarios the profile may be expected to not match
+  if (!IsSystemInManufacturingMode ()) {
+    ValidateActiveProfile ();
+  }
+
+  // Publish protocol for the configuration settings provider to be able to load with the correct profile in the PCD
+  Status = gBS->InstallProtocolInterface (
+                  &ImageHandle,
+                  (EFI_GUID *)&gConfProfileMgrProfileValidProtocolGuid,
+                  EFI_NATIVE_INTERFACE,
+                  NULL
+                  );
+
+  // if we fail to publish the protocol, the settings provider will not come up
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a failed to publish protocol (%r)!\n", __FUNCTION__, Status));
+    ASSERT (FALSE);
+  }
+
+  // Set the cached variable, in case of communication failure in the future
+  Status = gRT->SetVariable (
+                  CACHED_CONF_PROFILE_VARIABLE_NAME,
+                  &gConfProfileMgrVariableGuid,
+                  Attributes,

```

Comment:
There is a path to reach this that the attributes are un initialized.

Reply:
Thanks, will fix.